Message action sample widget (#63)
Esri/arcgis-experience-builder-sdk-resources: samples/widgets/message-subscriber/README.md

# Message subscriber

> This sample demonstrates how to use message action in widget.

Reviewer: ```suggestion
This sample demonstrates how to use message action in a widget.
```

## How to use the sample
Clone the [sample repo](https://github.com/esri/arcgis-experience-builder-sdk-resources) and copy this widget's folder (within `samples/widgets`) to the `client/your-extensions/widgets` folder of your Experience Builder installation.

## How it works

>   Within `query-action-setting.tsx`, create a class extends the React.PureComponent class with the types `ActionSettingProps`.

Reviewer: ```suggestion
  Within `query-action-setting.tsx`, create a class to extend the React.PureComponent class with the types `ActionSettingProps`.
```

  The configuration of the message action can be changed through the `onSettingChange` method.

  ```javascript
this.props.onSettingChange({
  actionId: this.props.actionId,
  config: this.props.config.set('fieldName', field['name']).set('useDataSource',{
    dataSourceId: this.props.config.useDataSource.dataSourceId,
    mainDataSourceId: this.props.config.useDataSource.mainDataSourceId,
    dataViewId: this.props.config.useDataSource.dataViewId,
    rootDataSourceId: this.props.config.useDataSource.rootDataSourceId,
    fields: allSelectedFields.map(f => f.jimuName)
  })
});

```

> Within `query-action.tsx`, create a class extends the `AbstractMessageAction` class. Then set the URI of the action setting component through the `getSettingComponentUri` method. Use the `filterMessageType` method to filter out actions that are not relevant to this widget. The `onExecute` method is called when the action is triggered.

Reviewer: ```suggestion
Within `query-action.tsx`, create a class to extend the `AbstractMessageAction` class. Then set the URI of the action setting component through the `getSettingComponentUri` method. Use the `filterMessageType` method to filter out actions that are not relevant to this widget. The `onExecute` method is called when the action is triggered.
```
